packages/react-refresh/src/ReactFreshRuntime.js

```diff
@@ -128,7 +128,7 @@ function haveEqualSignatures(prevType, nextType) {
   const nextSignature = allSignaturesByType.get(nextType);
 
   if (prevSignature === undefined && nextSignature === undefined) {
-    return true;
+    return false;
```

Comment:
This change would cause every component without Hooks to always be remounted. That's not right.